walkthrough/05-aws/04-reusing-stacks/README.md

```diff
@@ -0,0 +1,179 @@
+# 03 - Reusing Stacks
+
+This chapter will detail how Stacks themselves can be re-used by Stacks to reproduce a set of infrastructure instead of just reproducing individual Units.
```

Comment:
Why are we jumping right to this advanced use case of recursive stacks...before we've even shown inputs, or really played around with the stacks lifecycle at all! 

Before getting into recursion, I'd expect you to walk through:

- Anatomy of a stack
   - New blocks like `unit`
   - How you define a unit template
   - How you define a stack
   - How you reuse a stack
   - How you pass input vals
- What happens when you update a stack definition
- - What happens when you update a stack instance configuration
- What happens when you destroy a stack

Reply:
Ohh...you're using recursion to introduce the concept of a stack definition. Ok, I think what's throwing me off here is that I'm thinking in terms of _how I want to solve the problem_ whereas you're writing to showcase one feature at a time. So when the feature you intro seems "incomplete" to me, I get confused.

I *love* that I can look through the file system that corresponds to your README; that's a really neat idea, but maybe we should treat this as an "advanced" walkthrough and create a separate "get an idea of how it works" walkthrough that introduces the concepts in the following order:

- Plain old Terraform/Tofu
- Terragrunt Units 
- Terragrunt stacks with a unique definition per stack and no inputs
- Terragrunt stacks with a shared definition
- Terragrunt stacks with inputs

Reply:
That's chapters 1-4, right?

https://github.com/yhakbar/terragrunt-3313-stacks-walkthrough/tree/main/walkthrough